pytorch_lightning/loops/fit_loop.py

```diff
@@ -132,8 +145,8 @@ def done(self) -> bool:
         or if the maximum number of steps or epochs is reached.
         """
         # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
-        stop_steps = self.max_steps is not None and self.global_step >= self.max_steps
-        stop_epochs = self.max_epochs is not None and self.current_epoch >= self.max_epochs
+        stop_steps = FitLoop._is_max_limit_enabled(self.max_steps) and self.global_step >= self.max_steps
+        stop_epochs = FitLoop._is_max_limit_enabled(self.max_epochs) and self.current_epoch >= self.max_epochs
```

Comment:
```suggestion
        stop_steps = self._is_max_limit_enabled(self.max_steps) and self.global_step >= self.max_steps
        stop_epochs = self._is_max_limit_enabled(self.max_epochs) and self.current_epoch >= self.max_epochs
```

Reply:
Currently `_is_max_limit_enabled` is a static method of `FitLoop`. This doesn't could be both `self._is_max_limit_enabled` for both `FitLoop` and `checkpoint_connector`.

Reply:
@tchaton could you please take another look at ^

Reply:
If it used in 2 places, might be worth to make it an utility as it might not belong to FitLoop.

Reply:
@tchaton where would be a good file to keep a common util?

Reply:
maybe in the utilities file under pl/loops/utilities.py

Reply:
The function is going to be removed shortly, i don't think this is necessary